framework/include/executioners/Executioner.h

```diff
@@ -109,6 +109,141 @@ class Executioner : public MooseObject,
    */
   virtual bool lastSolveConverged();
 
+  class PicardSolve : public MooseObject, public PerfGraphInterface
```

Comment:
This needs to be in its own file - probably in the executioners directory

Reply:
Sure.